File: BB.Core/Model/Address.cs
namespace BB.Core.Model
{
    public class Address
    {
        public long UserID { get; set; }

        public string City { get; set; }

        public string Country { get; set; }

        public string Street { get; set; }

        public string HouseNumber { get; set; }

        public int PostalCode { get; set; }

        public virtual UserDetails UserDetails { get; set; }
    }
}

File: BB.Web/app/Model/Address/Address.ts
import ud = require("../Users/UserDetails");
import UserDetails = ud.UserDetails;

export class Address {
    public UserID: number;
    public City: string;
    public Country: string;
    public Street: string;
    public HouseNumber: string;
    public PostalCode: number;
    public UserDetails: UserDetails;
}